File: 05_arrays/05_24_merge_sorted_arrays/merge_sorted_arrays.js
const merge = (nums1 = [], m, nums2 = [], n) => {
    let i = m - 1
    let j = n - 1
    let k = m + n - 1

    while (i >= 0 && j >= 0) {
        nums1[k--] = nums1[i] > nums2[j] ? nums1[i--] : nums2[j--]
    }
    while (j >= 0) {
        nums1[k--] = nums2[j--]
    }
}

let nums1 = [1, 2, 3, 0, 0, 0];
let m = 3;
let nums2 = [2, 5, 6];
let n = 3;

merge(nums1, m, nums2, n);

console.log("Merged Array:", nums1.join(" "));

File: 05_arrays/05_24_merge_sorted_arrays/merge_sorted_arrays.cpp
#include <bits/stdc++.h>

using namespace std;

void merge(vector<int> &nums1, int m, vector<int> &nums2, int n)
{
    int temp = m + n - 1;
    int i = m - 1;
    int j = n - 1;
    while (i >= 0 && j >= 0)
    {
        nums1[temp--] = nums1[i] < nums2[j] ? nums2[j--] : nums1[i--];
    }
    while (j >= 0)
    {
        nums1[temp--] = nums2[j--];
    }
}

int main()
{
    vector<int> nums1 = {1, 2, 3, 0, 0, 0};
    int m = 3;
    vector<int> nums2 = {2, 5, 6};
    int n = 3;

    merge(nums1, m, nums2, n);

    cout << "Merged Array: ";
    for (int num : nums1)
    {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}